pkg/core/transaction/transaction.go

```diff
@@ -117,6 +117,17 @@ func (t *Transaction) Hash() util.Uint256 {
 	return t.hash
 }
 
+// SetHash updates Transaction hash. This method directly modifies Transaction's
+// hash without any respect to the Transaction's content. The resulting hash is
+// being cached, hence subsequent calls to Hash will return exactly the same
+// value without an attempt to serialize Transaction. This method is designated
+// for `invokecontainedscript` RPC API users only and MUST NOT be used for other
+// purposes.
+func (t *Transaction) SetHash(hash util.Uint256) {
```

Comment:
Same with these methods.

Reply:
You have to specify this hash on the client side somehow. And if we don't want to introduce additional wrappers over `transaction.Transaction` struct, then it looks like the only way to me.

Reply:
Can't say it's elegant. Maybe some additional constructor? `NewFakeTransaction(script, hash, size)`?

Reply:
Fixed.